lib/scope.js

```diff
@@ -255,13 +255,13 @@ AST_Lambda.DEFMETHOD("init_scope_vars", function(){
     this.variables.set(symbol.name, def);
 });
 
-AST_SymbolRef.DEFMETHOD("reference", function() {
+AST_SymbolRef.DEFMETHOD("reference", function(options) {
     var def = this.definition();
     def.references.push(this);
     var s = this.scope;
     while (s) {
         push_uniq(s.enclosed, def);
-        if (s === def.scope) break;
+        if (s === def.scope && !options.keep_fnames) break;
```

Comment:
Can you please add a comment above this line why the `&& !options.keep_fnames` clause is important. Otherwise it may be inadvertently removed in the future.